services/121-service/test/program/manage-program-financial-service-provider-configuration.test.ts

```diff
@@ -193,11 +198,48 @@ describe('Manage financial service provider configurations', () => {
     expect(getResultConfig).toBeUndefined();
   });
 
+  it('should not delete existing program financial service provider configuration because of active registrations with that config', async () => {
+    // Prepare
+    await seedPaidRegistrations([registrationOCW5], programIdVisa);
+
+    // Act
+    const name = seededFspConfigVoucher.financialServiceProvider;
+    const result = await deleteProgramFinancialServiceProviderConfiguration({
+      programId: programIdVisa,
+      name,
+      accessToken,
+    });
+    const getResult = await getProgramFinancialServiceProviderConfigurations({
+      programId: programIdVisa,
+      accessToken,
+    });
+    const getResultConfig = getResult.body.find(
+      (config) => config.name === name,
+    );
+
+    // Assert
+    expect(result.statusCode).toBe(HttpStatus.CONFLICT);
+    expect(result.body).toMatchSnapshot();
+    expect(getResultConfig).toBeDefined();
+  });
+
   // Checking this exception in api test because it's hard to unit test the more complex transaction querybuilder part
-  it('should not delete existing program financial service provider configuration because of transactions', async () => {
+  it('deleting program financial service provider configuration with existing transactions should set programFinancialServiceProviderConfigurationId of transactions to null', async () => {
```

Comment:
so this test fails apparently because it runs into 409 instead of 204 despite the fact that you deleted them first. That needs to be checked and fixed.

Reply:
I did not wait for the status change to be completed. Therefore it failed inconsistently here vs local